data/rewards/calculator.go

```diff
@@ -108,17 +120,29 @@ func (calc *RewardCalculator) Calculate() (amt *balance.Amount, err error) {
 }
 
 func (calc *RewardCalculator) secondsPerCycleLatest() (int64, time.Time) {
-	tEnd := calc.blockStore.LoadBlockMeta(1).Header.Time.UTC()
+	tEnd := calc.genesisTime.UTC()
 	secsPerCycle := calc.options.EstimatedSecondsPerCycle
 	if calc.height > calc.options.BlockSpeedCalculateCycle {
 		// get speed calculation [begin, end] height
 		cycle := calc.options.BlockSpeedCalculateCycle
-		cycleEndHeight := (calc.height-1)/cycle*cycle + 1
+		cycleEndHeight := calc.height
```

Comment:
Are you going to assume calc.height == cycle_end_height? Have you tested kill/restart a running node in which case cache goes away after restart?

Reply:
I can revert this back to the way it was just to be safe.